rusk/src/bin/config/wallet.rs

```diff
@@ -5,72 +5,50 @@
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
 use clap::{App, Arg, ArgMatches};
+use dusk_bytes::DeserializableSlice;
+use dusk_pki::PublicSpendKey;
 use rusk::Result;
 use serde::{Deserialize, Serialize};
-use std::fs;
+use tracing::warn;
 
-#[derive(Serialize, Deserialize, Clone)]
+#[derive(Serialize, Deserialize, Default, Clone)]
 pub(crate) struct WalletConfig {
-    pub(crate) path: String,
-    password: Option<String>,
-    password_path: Option<String>,
+    generator_key: Option<String>,
 }
 
-/// Default wallet path.
-pub const WALLET_PATH: &str = "wallet.dat";
-
-impl Default for WalletConfig {
-    fn default() -> Self {
-        WalletConfig {
-            path: WALLET_PATH.to_string(),
-            password: None,
-            password_path: None,
-        }
-    }
+fn parse_generator_key(key: &str) -> Result<PublicSpendKey> {
+    let key = bs58::decode(key).into_vec()?;
+    let key = PublicSpendKey::from_slice(&key)?;
+    Ok(key)
 }
 
 impl WalletConfig {
-    pub fn _password(&self) -> Result<String> {
-        if let Some(password) = &self.password {
-            return Ok(password.to_string());
-        };
-        if let Some(password_path) = &self.password_path {
-            return Ok(fs::read_to_string(password_path)?);
-        };
-        panic!("No <password> nor <password_path> are provided>");
+    pub(crate) fn generator(&self) -> Option<PublicSpendKey> {
+        // Unwrapping here is ok since we ensure the key is valid when ingesting
+        // it.
+        self.generator_key
+            .as_ref()
+            .map(|key| parse_generator_key(key).unwrap())
```

Comment:
If you have the change described above you don't need to `unwrap` either. You could just have something like:
```rust
        parse_generator_key(self.generator_key.as_deref()?)
```